src/Admin/ReaderThemes.php

```diff
@@ -140,12 +143,27 @@ static function( $theme ) {
 
 		$themes = array_map(
 			function ( $theme ) {
+				$theme                 = $this->normalize_theme_data( $theme );
```

Comment:
Are you sure`normalize_theme_data` needs to run again? Unless I'm missing something, it seems to be applied to all the themes before this point.

Reply:
Yes, as it will make sure that any `WP_Theme` objects discovered after looking for themes with the `AMP` header will also get converted into arrays, right?

Reply:
Isn't that already happening on line 97, though?

Reply:
Ah, right.. I meant that it is actually needed for the themes that get passed through the `amp_reader_themes` filter.

Reply:
Because someone could push a theme onto the list of `amp_reader_themes` and it can then lack the normalized fields wirthout this.

Reply:
Ok, that makes sense. It still feels funny to run the normalize function twice on all the default themes and compatible themes. Maybe it could run here only and not in the other places.

Reply:
It does feel weird, but it's important for the themes being passed _into_ the `amp_reader_themes` filter be all arrays and not intermixed with `WP_Theme` objects.

Reply:
Oh, right, got it. Approving.